packages/core/src/client.ts

```diff
@@ -2771,12 +2771,12 @@ export class MedplumClient extends TypedEventTarget<MedplumClientEventMap> {
   }
 
   /**
-   * Returns true if the client is waiting for authentication.
-   * @returns True if the client is waiting for authentication.
+   * Returns true if the client is waiting for initial authentication.
+   * @returns True if the client is waiting for initial authentication.
    * @category Authentication
    */
   isLoading(): boolean {
-    return !this.isInitialized || !!this.profilePromise;
+    return !this.isInitialized || (!!this.profilePromise && !this.sessionDetails?.profile);
```

Comment:
The `!!` mixed in with the `!`s threw me a bit when scanning this line — minor suggestion to clarify with e.g.:
```suggestion
    return !this.isInitialized || (Boolean(this.profilePromise) && !this.sessionDetails?.profile);
```